### DISCUSSION TITLE
Matching for multiple device names in per-device hardware settings

### DISCUSSION THREAD

**Discussion by SirWrexes** [hyprwm/Hyprland#2650] [Q&A]
When looking up `hyprctl devices` in my terminal, I see my keyboards being split in different devices.
![image](https://github.com/hyprwm/Hyprland/assets/5567164/5e7504e0-1a58-4e9e-b83c-b4ee8cd5b8fd)

I don't exactly know why it does that, and quite frankly I've tested setting just one and it worked... but I still wanted to make sure everything is consistent and c/p'd settings for all parts, and it left me with a question:
**_Is there a way to match multiple device names as a group for per-device settings ?_**

I've tried all of these
```cfg
# Ignores the device with -1 suffix, so no partial name matching
device:cherry-gmbh-cherry-corded-device {
 ...
}

# Doesn't work at all, so no globbing either
device:cherry-gmbh-* { 
  ...
} 

# Ignores the first device, with and without comma at the end of the first line
device:cherry-gmbh-cherry-corded-device,
device:cherry-gmbh-cherry-corded-device-1 {
 ...
}

# Ignores both devices, with and without separator comma
device:cherry-gmbh-cherry-corded-device, device:cherry-gmbh-cherry-corded-device-1 {
 ...
}
```

Sure, I could c/p settings for split devices, but just look at this Thinkpad's keyboard, it's all over the place: 5 devices for one keyboard. So it's 5 c/ps for one piece of equipment that's embedded in the PC.
 I could use global settings, but then the French mapping would override my UK Cherry's keymap as mentioned in the info bit of [that section of the docs](https://wiki.hyprland.org/Configuring/Keywords/#per-device-input-configs), and my goal initally was to avoid needing a keymap switcher bind. :thinking:

**Comment by SirWrexes** [comment]
As a side note, a way I found to reduce copy/pasting a little bit and to ensure my config changes replicate among all parts of my devices is to make a separate file (for instance `hardware.thinkpad-keyboard.conf`), have my `kb_*` & co settings in there,
and then in my main hardware config file I do this
```cfg
device:<device_name_1> {
  source = hardware.thinkpad-keyboard.conf
}

device:<device_name_2> {
  source = hardware.thinkpad-keyboard.conf
}

device:<device_name_3> {
  source = hardware.thinkpad-keyboard.conf
}

...
```

It works, but is still not ideal.

**Comment by visch** [comment]
I have this same problem right now, glob syntax would be pretty nice here!
